test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodControllerSpec.scala

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers
+
+import play.api.mvc.{Action, AnyContent}
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.GetSelfEmploymentsHttpParser.UnexpectedStatusFailure
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.httpparser.PostSelfEmploymentsHttpParser.PostSelfEmploymentsSuccessResponse
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.connectors.mocks.MockIncomeTaxSubscriptionConnector
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.BusinessAccountingMethodForm
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models.AccountingMethodModel
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.utilities.TestModels._
+
+class BusinessAccountingMethodControllerSpec extends ControllerBaseSpec
+  with MockIncomeTaxSubscriptionConnector {
+
+  override val controllerName: String = "BusinessAccountingMethodController"
+  override val authorisedRoutes: Map[String, Action[AnyContent]] = Map(
+    "show" -> TestBusinessAccountingMethodController$.show(),
+    "submit" -> TestBusinessAccountingMethodController$.submit()
+  )
+
+  object TestBusinessAccountingMethodController$ extends BusinessAccountingMethodController(
+    mockMessagesControllerComponents,
+    mockIncomeTaxSubscriptionConnector,
+    mockAuthService
+  )
+
+  def modelToFormData(accountingMethodModel: AccountingMethodModel): Seq[(String, String)] = {
+    BusinessAccountingMethodForm.businessAccountingMethodForm.fill(accountingMethodModel).data.toSeq
+  }
+
+  "Show" should {
+
+    "return ok (200)" when {
+      "the connector returns data" in {
+        mockAuthSuccess()
+        mockGetSelfEmployments(BusinessAccountingMethodController.businessAccountingMethodKey)(
+          Right(Some(testAccountingMethodModel))
+        )
+        val result = TestBusinessAccountingMethodController$.show()(FakeRequest())
+        status(result) mustBe OK
+        contentType(result) mustBe Some("text/html")
+      }
+      "the connector returns no data" in {
+        mockAuthSuccess()
+        mockGetSelfEmployments(BusinessAccountingMethodController.businessAccountingMethodKey)(Right(None))
+        val result = TestBusinessAccountingMethodController$.show()(FakeRequest())
+        status(result) mustBe OK
+        contentType(result) mustBe Some("text/html")
+      }
+    }
+    "Throw an internal exception error" when {
+      "the connector returns an error" in {
+        mockAuthSuccess()
+        mockGetSelfEmployments(BusinessAccountingMethodController.businessAccountingMethodKey)(Left(UnexpectedStatusFailure(INTERNAL_SERVER_ERROR)))
+        intercept[InternalServerException](await(TestBusinessAccountingMethodController$.show()(FakeRequest())))
```

Comment:
This test should probably check the error message we are throwing.

Reply:
fixed